### PR TITLE
Implement absolute_import in Scrapers w/ tests

### PR DESCRIPTION
**Pull request by cclauss**
Partial fix for #226
This should help the codebase to achieve Python 3 compatibility.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `master` branch.
- [X] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Adds .imports within a module
- Adds several tests to test_server.py

### CODE REVIEW

**Comment by cclauss** [comment]
Codacy does not like the asserts that drive Pytest
* https://github.com/fossasia/query-server/pull/244#issuecomment-338281076

**Comment by cclauss** [comment]
@bhaveshAn @niranjan94 Please review.... I believe that I have __FINALLY__ figured out how to resync with upstream.  The key learning tool for me was:
* https://docs.python.org/devguide/gitbootcamp.html#syncing-with-upstream

**Comment by cclauss** [comment]
;-)  good catch...  Each _item_ is an ascii character converted into an int.  I will dig in.  It is good to see that at least we can load the scrapers in Py3 for the first time even if they do not provide the desired output.  Also good to see the pytests kicking in too...
* https://travis-ci.org/fossasia/query-server/jobs/295395016#L549

Did Py2 do OK or did your testing not get that far?

**Comment by bhaveshAn** [comment]
Python2 worked okay :smile:

**Comment by cclauss** [comment]
Thanks.  Good news.  That means that this PR does no harm.

**Comment by cclauss** [comment]
@bhaveshAn Checking this branch for Python 3 compatibility is not in scope.  We have removed our to commitment to Python 3 from the README.  This PR is designed to do one thing: to do relative imports.  It does that.  Now. I have tried to fix a problem that was not required to fix and I am left with a PR that has two commits because I rebased back too far in history and I can not git reset --hard HEAD by way out of it.

**Comment by cclauss** [comment]
This PR is ready for review.
* Python 3 is out of scope given our README changes.
* The pytests pass and will help with future PRs
* Codacy fails because it does not like when we add assert statements but the are valid Python

**Comment by bhaveshAn** [comment]
@cclauss Have you checked it with Python2 ??

**Comment by cclauss** [comment]
@bhaveshAn Yes.  Are you seeing something strange on your end?

**Comment by bhaveshAn** [comment]
All I mean is if we removed support for Python3 from README than we don't need this PR, as python2 is working without this change.

**Comment by cclauss** [comment]
Do we want to avoid Python 3 forever?

**Comment by aashutoshrathi** [comment]
@bhaveshAn Please rebase your PR.
First update your fork's master and pull it, then rebase current branch with master.

**Comment by bhaveshAn** [comment]
> @bhaveshAn Please rebase your PR.
First update your fork's master and pull it, then rebase current branch with master.

@aashutoshrathi I am not having any open PR right now. This PR is of @cclauss :sweat_smile:

**Comment by aashutoshrathi** [comment]
Oh, my bad, GitHub suggested and I didn't checked again. 😅

**Comment by bhaveshAn** [comment]
No problem @aashutoshrathi

**Comment by cclauss** [comment]
@aashutoshrathi Your review of these changes?

**Comment by aashutoshrathi** [comment]
@cclauss you need to resolve conflicts in `app/test_server.py` and then merge.

**Comment by cclauss** [comment]
Rebased and squashed again.

**Comment by cclauss** [comment]
Rebased and squashed yet again. @aashutoshrathi Please review these changes

**Comment by cclauss** [comment]
@aashutoshrathi please read https://github.com/fossasia/query-server/pull/297#issuecomment-340312864 above to understand why ignoring Codasy's error messages related to assert statements increase code quality.

**Comment by cclauss** [comment]
Is it useful continually re-base my pull requests?

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by cclauss** [comment]
Which conflicts?

File: app/test_server.py [hunk 1]
```diff
@@ -24,20 +16,75 @@ def test_true():
...
+@pytest.mark.xfail(reason='small_test() needs to be fixed.')
+def test_small_test():
+    small_test()
+
+
+@pytest.mark.xfail
 def test_invalid_url_api_call():
     response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
     assert response.json()['Status Code'] == 404
+    assert response.status_code == response.json()['Status Code']
```
Comment (AnshulMalik) [inline app/test_server.py]: how is this helping ?

File: app/test_server.py [hunk 2]
```diff
@@ -24,20 +16,75 @@ def test_true():
...
     assert requests.get(url).json()['Status Code'] == 400, engine_name
 
+    if engine_name in ('exalead', 'mojeek'):
+        return
+    fmt = 'http://localhost:7001/api/v1/search/{}?query=fossaisa&num=5'
+    response = requests.get(fmt.format(engine_name))
+    assert response.status_code == 200, '{}: {}'.format(engine_name,
+                                                        response.status_code)
+    links = response.json()
+    if engine_name not in ('quora', 'tyoutube'):
```
Comment (AnshulMalik) [inline app/test_server.py]: I didn't get this `if`

File: app/test_server.py [hunk 3]
```diff
@@ -24,20 +16,75 @@ def test_true():
...
     response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
     assert response.json()['Status Code'] == 404
+    assert response.status_code == response.json()['Status Code']
 
 
 def make_engine_api_call(engine_name):
     url = 'http://localhost:7001/api/v1/search/' + engine_name
     assert requests.get(url).json()['Status Code'] == 400, engine_name
 
+    if engine_name in ('exalead', 'mojeek'):
```
Comment (AnshulMalik) [inline app/test_server.py]: Not testing for these?

File: app/test_server.py (hunk 1 above)
Comment (cclauss) [reply app/test_server.py]: Good catch.  I will use the first form in the test above and then remove this test.

File: app/test_server.py (hunk 2 above)
Comment (cclauss) [reply app/test_server.py]: I will add a comment to explain.

File: app/test_server.py (hunk 3 above)
Comment (cclauss) [reply app/test_server.py]: I will add a comment to explain.  Thanks for your review.

**Comment by bhaveshAn** [comment]
Update

**Comment by cclauss** [comment]
@bhaveshAn Can you please review?

**Pull request closed without merging**